File: algorithms/solutions/week22/toanha/majority-element.go
package main

func majorityElement(nums []int) int {
	n := len(nums) / 2
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		m[nums[i]]++
		if value, ok := m[nums[i]]; ok {
			if value > n {
				return nums[i]
			}
		}
	}

	return -1
}

File: algorithms/solutions/week22/unrealhoang/majority_element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
    count = {}
    nums.each do |num|
        count[num] ||= 0
        count[num] += 1
    end
    count.max_by { |k, v| v }[0]
end
